Added Jupyter Notebook Back to the code

Prakashlokesh opened the pull request.

puravparab reviewed (changes requested):
This should be either:
```
ENTRYPOINT ["python"]
CMD ["app.py"]
```
or
```
CMD ["python", "app.py"]
```

Add jupyter as a subprocess in app.py

Prakashlokesh commented:
Purav can you check once now?

puravparab commented on Dockerfile:
```diff
@@ -47,17 +47,20 @@ COPY requirements.txt /app/requirements.txt
 # Install Python packages from requirements.txt
 RUN mamba install --yes --file requirements.txt && mamba clean --all -f -y
-RUN pip install rank_bm25 streamlit-pdf-viewer
+RUN pip install rank_bm25 streamlit-pdf-viewer jupyter
```
Take out jupyter from here. We don't need this since it's present in our requirements.txt

puravparab commented:
Add this to app.py:
```
if __name__ == "__main__":
    # If streamlit instance is running
    if os.environ.get("STREAMLIT_RUNNING") == "1":
        main()
    else:
        os.environ["STREAMLIT_RUNNING"] = "1"  # Set the environment variable to indicate Streamlit is running
        subprocess.Popen(["streamlit", "run", __file__, "--server.port=5003", "--server.address=0.0.0.0", "--server.baseUrlPath=/team3"])
        subprocess.run(["jupyter", "notebook", "--ip=0.0.0.0", "--port=6003", "--no-browser", "--allow-root", "--NotebookApp.base_url=/team3/jupyter"])
```

Add update the Setup section in README.md to include port 6003

Prakashlokesh commented:
Purav I have updated the code given by you in app.py and also updated the readme file. Please let me know if you face any conflicts...

![Screenshot 2024-10-23 at 8 17 09 PM](https://github.com/user-attachments/assets/5aa4434e-6a53-4559-9c4d-cf72350a4235)

Prakashlokesh commented:
Hi Purav,

I have update the code as per your requirements. Please go through it and pplease let me know if you find any conflicts.
